doc/build/cmake/style.rst

```diff
@@ -0,0 +1,126 @@
+:orphan:
+
+.. _cmake-style:
+
+CMake Style Guidelines
+======================
+
+General Formatting
+------------------
+
+- **Indentation**: Use **2 spaces** for indentation. Avoid tabs to ensure consistency across different environments.
+- **Line Length**: Limit line length to **100 characters** where possible.
+- **Empty Lines**: Use empty lines to separate logically distinct sections within a CMake file.
+- **No Space Before Opening Brackets**: Do not add a space between a command and the opening parenthesis.
+  Use ``if(...)`` instead of ``if (...)``.
+
+  .. code-block:: cmake
+
+     # Good:
+     if(ENABLE_TESTS)
+         add_subdirectory(tests)
+     endif()
+
+     # Bad:
+     if (ENABLE_TESTS)
+         add_subdirectory(tests)
+     endif()
+
+Commands and Syntax
+-------------------
+
+- **Lowercase Commands**: Always use **lowercase** CMake commands (e.g., ``add_executable``, ``find_package``). This improves readability and consistency.
+
+  .. code-block:: cmake
+
+     # Good:
+     add_library(my_lib STATIC src/my_lib.cpp)
+
+     # Bad:
+     ADD_LIBRARY(my_lib STATIC src/my_lib.cpp)
+
+- **One Argument per Line**: Break the file arguments across multiple lines to make it easier
+  to scan and identify each source file or item.
+
+  .. code-block:: cmake
+
+     # Good:
+     target_sources(my_target PRIVATE
+         src/file1.cpp
+         src/file2.cpp
```

Comment:
this uses four space indent.

Btw. do we want just the indent or alignment ?
I'm fine either way, but prefer consistency.

And should function calls spanning multiple lines wrap first arg also, that is if we only indent and not align the args.

A) 
```
target_sources(my_target PRIVATE
  src/file1.cpp
  src/file2.cpp
)
```
A.a)
```
target_sources(
  my_target PRIVATE
  src/file1.cpp
  src/file2.cpp
)
```

B)
```
target_sources(my_target PRIVATE
               src/file1.cpp
               src/file2.cpp
)
```